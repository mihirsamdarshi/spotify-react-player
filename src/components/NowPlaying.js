import React, { useState } from 'react';
import { Card, CardContent, IconButton, Typography } from '@material-ui/core';
import SkipPreviousIcon from '@material-ui/icons/SkipPrevious';
import PlayArrowIcon from '@material-ui/icons/PlayArrow';
import SkipNextIcon from '@material-ui/icons/SkipNext';
import PauseIcon from '@material-ui/icons/Pause';
import '../stylesheets/NowPlaying.scss';
import { nextSong, pauseSong } from '../scripts/api';
import { returnArtistAlbumString } from '../scripts/helpers';

const NowPlaying = (props) => {
    const nowPlayingCard = {
        backgroundImage: `url(${props.nowPlaying.img.url})`,
        backgroundPosition: 'center',
        maxWidth: '100%',
        height: 'auto',
    };

    const [songPlaying, setSongPlaying] = useState(false);

    const handlePlay = () => {
        if (songPlaying) {
            pauseSong();
            setSongPlaying(false);
        } else {
            console.log('play func here');
            setSongPlaying(true);
        }
    };

    const handlePrevious = () => {
        previousSong()
    };

    const handleNext = () => {
        nextSong();
    };

    return (
        <Card>
            <div className="bg-image" style={nowPlayingCard} />
            <div className="nowPlayingDetails">
                <CardContent className="nowPlayingContent">
                    <Typography component="h5" variant="h5">
                        {props.nowPlaying.name}
                    </Typography>
                    <Typography variant="subtitle1" color="textSecondary">
                        {returnArtistAlbumString(props.nowPlaying)}
                    </Typography>
                </CardContent>
                <div className="nowPlayingControls">
                    <IconButton aria-label="previous" onClick={handlePrevious}>
                        {<SkipPreviousIcon />}
                    </IconButton>
                    <IconButton aria-label="play/pause" onClick={handlePlay}>
                        {songPlaying ? <PauseIcon /> : <PlayArrowIcon />}
                    </IconButton>
                    <IconButton aria-label="next" onClick={handleNext}>
                        {<SkipNextIcon />}
                    </IconButton>
                </div>
            </div>
        </Card>
    );
};

export default NowPlaying;
