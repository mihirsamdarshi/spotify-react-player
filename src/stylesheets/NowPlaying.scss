.bg-image {
  height: 100%;
  width: 100%;
}
.nowPlayingDetails {
  background-color: rgb(0,0,0); /* Fallback color */
  background-color: rgba(0,0,0, 0.4); /* Black w/opacity/see-through */
  display: flex;
  flex-direction: row;
  z-index: 2;
  text-align: center;
}

.nowPlayingContent {
  display: flex;
  flex: 1 0 auto;
  flex-direction: column;
}

.nowPlayingControls {
  display: flex;
  align-items: center;
}

.playIcon {
  height: 100px;
  width: 100px;
}
