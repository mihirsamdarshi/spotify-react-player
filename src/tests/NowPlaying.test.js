import React from 'react';
import ReactDOM from 'react-dom';
import NowPlaying from '../components/NowPlaying';

const fakeNowPlayingData = {
    album: 'Test - The LP',
    img: {
        url: 'test.com',
    },
    artist: [
        {
            name: 'Testy McTestFace',
        },
    ],
};

it('renders the Now Playing component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NowPlaying nowPlaying={fakeNowPlayingData} />, div);
    ReactDOM.unmountComponentAtNode(div);
});
