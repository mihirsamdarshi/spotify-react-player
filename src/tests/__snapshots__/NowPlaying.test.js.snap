// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should start with fake now playing data 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
>
  <div
    className="bg-image"
    style={
      Object {
        "backgroundImage": "url(test.com)",
        "backgroundPosition": "center",
        "height": "auto",
        "maxWidth": "100%",
      }
    }
  />
  <div
    className="nowPlayingDetails"
  >
    <div
      className="MuiCardContent-root nowPlayingContent"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      />
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        Testy McTestFace
      </h6>
    </div>
  </div>
</div>
`;
